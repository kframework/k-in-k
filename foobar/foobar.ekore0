module FOOBAR
  sort Buzz { } [ ]

  syntax Foo ::= "bar" [klabel(bar), function]
               | "buzz" [klabel(buzz)]
  rule `bar`() => `buzz`()
endmodule

