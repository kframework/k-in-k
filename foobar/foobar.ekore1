module FOOBAR
  sort Buzz { } [ ]

  syntax Foo ::= "bar" [klabel(bar), function]
               | "buzz" [klabel(buzz2)]
  rule bar{}() => buzz{}()
endmodule

