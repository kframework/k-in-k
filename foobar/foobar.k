module FOOBAR
  syntax Foo ::= "bar" [klabel(bar), function]
               | "buzz" [klabel(buzz)]
  rule bar => buzz
endmodule

