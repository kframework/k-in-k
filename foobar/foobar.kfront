kmodule FOOBAR
  // K-BASIC + KSEQ
  ksyntax( ksort(SortK)
         , klabel(dotk)
         , // args
         , // attrs
         )

  // No concrete KItems are declared, but we need the KItem sort to exist for kseq Sort
  // TODO: Allow `ksyntax(ksort(SortKItem))`
  ksyntax( ksort(SortKItem)
         , klabel(dummy)
         , // args
         , // attrs
         )

  ksyntax( ksort(SortK)
         , klabel(kseq)
         , ksort(SortKItem) ; ksort(SortK)
         , // attrs
         )

 // bar
  ksyntax( ksort(SortFoo)
         , klabel(Lblbar'Unds'FOOBAR'Unds')
         ,
         , kattribute(function)
         )

  // buzz
  ksyntax( ksort(SortFoo)
         , klabel(Lblbuzz'Unds'FOOBAR'Unds')
         , // args
         ,
         )
  krule( klabel(Lblbar'Unds'FOOBAR'Unds')
       , klabel(Lblbuzz'Unds'FOOBAR'Unds')
       )
endkmodule
