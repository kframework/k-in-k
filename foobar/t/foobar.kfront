ksyntax(ksort("Foo"), klabel("bar" ))
ksyntax(ksort("Foo"), klabel("buzz"))
