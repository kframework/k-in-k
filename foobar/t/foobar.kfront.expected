<kink>
  ksyntax ( ksort ( "Foo" ) , klabel ( "bar" ) )  ksyntax ( ksort ( "Foo" ) , klabel ( "buzz" ) )  .KFrontDeclarations
</kink>
